feat: add employee categories management interface

ivelin-web opened the pull request:
- Add employment types management page with create/delete functionality
- Implement API services for fetching, creating, and deleting employment types
- Add TypeScript types for employment type operations
- Create responsive UI with modal for adding new types
- Show system vs custom type indicators
- Add confirmation dialog for deletion
- Integrate with existing notification and confirmation systems

ivelin-web committed:
feat: add employee categories management interface

- Add employment types management page with create/delete functionality
- Implement API services for fetching, creating, and deleting employment types
- Add TypeScript types for employment type operations
- Create responsive UI with modal for adding new types
- Show system vs custom type indicators
- Add confirmation dialog for deletion
- Integrate with existing notification and confirmation systems

ivelin-web committed:
fix: support custom employment types in leave type creation

- Update validation schema to accept dynamic employment type keys instead of hardcoded ones
- Modify service to fetch all employment types (global + organization-specific)
- Remove dependency on hardcoded EMPLOYMENT_TYPE_VALUES constant
- This allows creating leave types with custom employment categories like "Test"

ivelin-web committed:
Improve employment types badge styling with professional design

- Enhanced badge styling with gradient backgrounds and rounded corners
- Added subtle hover effects and transitions
- Improved typography hierarchy and spacing
- Added calendar icon for better visual hierarchy
- Professional date formatting
- Better layout structure and consistent spacing

deegha commented on src/views/organization/employmentTypes.tsx:
```diff
@@ -0,0 +1,194 @@
+      actionButton={<Button onClick={() => setIsCreateModalOpen(true)}>Add New Type</Button>}
+      pageName="Employment Types"
+    >
+      {isLoading ? (
```
Please create a shimmer for this as we have used in the other places.

deegha commented on src/views/organization/employmentTypes.tsx:
```diff
@@ -0,0 +1,194 @@
+    });
+  };
+
+  if (error) return <div>Error loading employment types</div>;
```
lets create a generic reusable error component.

deegha commented on src/views/organization/employmentTypes.tsx:
```diff
@@ -0,0 +1,194 @@
+      )}
+
+      {/* Create Modal */}
+      {isCreateModalOpen && (
```
lets move this to a separate file.

ivelin-web committed:
Match Employment Types design with Leave Management - System badges and Custom delete icons

ivelin-web committed:
Refactor Employment Types page following PR feedback

- Replace simple loading text with EmploymentTypesShimmer component for consistent UI
- Create reusable ErrorDisplay component for proper error handling with real API messages
- Extract CreateEmploymentTypeModal to separate file for better code organization
- Update CLAUDE.md with conventions for loading and error states
